### DISCUSSION TITLE
Use Supabase API instead of Supabase JS

### DISCUSSION THREAD

**Discussion by LunatiqueCoder** [supabase/supabase#15563] [Questions] [answered]
Hello! 

I really love Supabase! 🚀 

I'm building an app with React and Redux Toolkit Query, and I was wondering if I could use Supabase API like this: 
<details><summary>👇 Click to expand example code </summary>

```typescript
import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';
import {SUPABASE_API_KEY, SUPABASE_URL} from 'src/constants';
import {RootState} from 'src/store';
import packageJson from '../../../package.json';

export const RPC = '/rest/v1/rpc';

const baseQuery = fetchBaseQuery({
  baseUrl: SUPABASE_URL,
  prepareHeaders: (headers, {getState}) => {
    const state = getState() as RootState;
    const token = state.auth.session?.access_token;

    headers.set('apikey', SUPABASE_API_KEY);
    headers.set('content-profile', 'public');

    headers.set(
      'x-client-info',
      `supabase-js/${packageJson.dependencies['@supabase/supabase-js']}`.replace(
        '^',
        '',
      ),
    );

    if (token) {
      headers.set('authorization', 'Bearer ' + token);
    }

    return headers;
  },
});


// Define a service using a base URL and expected endpoints
export const apiSlice = createApi({
  reducerPath: 'apiSlice',
  baseQuery: baseQuery,
  endpoints: builder => ({
    getUserProfile: builder.query<string, void>({
      query: () => ({
        url: `${RPC}/get_user_profile`,
        method: 'POST',
      }),
    }),
  }),
});

export const {useGetUserProfileQuery} = apiSlice;


```

</details>

Is it a good approach or perhaps a bad idea?

**Comment by GaryAustin1** [comment, marked as answer]
Not really following all of your code, but you can certainly use all of the Supabase API thru the REST API directly with http calls.  Setting the api-key and Authorization headers for RLS work fine.
https://postgrest.org/en/stable/ has the best info if you are making REST calls to the database directly on the URL and body requirements.

**Comment by LunatiqueCoder** [comment]
Great, thank you :) Just starting with Postgres for a pet project so I'm sorry if I'm asking newbie questions.

Basically, yeah, i was setting the api key and Authorization headers plus some extra headers that I saw Supabase JS was adding like `content-profile` and `x-client-info`,  but those docs seem to be exactly what I was looking for.
